Added strict validation

ywkim312 opened the pull request:
## Description
Added strict validation in Contributions building block and simplified the query.

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Immediately
- [ ] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

sandeep-ps commented:
Hi @ywkim312, I'm reviewing this PR. Sometimes when doing POST /contributions, the building block is returning HTTP 400 Bad Request. I have added one request body below. I'm using randomly generated names and descriptions to avoid name conflicts. When you get some time, could you please take a look at this? Thanks.
 
```
{
    "capabilities": [
        {
            "apiBaseUrl": null,
            "apiDocUrl": null,
            "dataDeletionEndpointDetails": {
                "apiKey": "apiKey",
                "description": "description",
                "deletionEndpoint": "deletionEndpoint"
            },
            "deploymentDetails": {
                "authMethod": null,
                "databaseDetails": null,
                "environmentVariables": null,
                "location": "internal"
            },
            "description": "Cupiditate esse aliquid tenetur illum et consequuntur consequatur. Et non vel ratione dolore totam suscipit numquam nemo. In vitae sed consequatur sunt sit et omnis id quia.
 
Rerum dicta a sint tenetur voluptatum voluptate. Nesciunt labore neque. Ducimus sunt qui asperiores quaerat facilis iste in. Minima dicta molestiae est laboriosam aut non corporis consectetur. Sit aperiam et accusamus tempore atque nam occaecati. Id nesciunt molestiae beatae sed sequi laudantium autem.
 
Accusamus magnam veritatis eius. Enim a dolorem quasi dolor neque voluptas debitis. Culpa est ut vero facere possimus. Quia facere velit voluptates hic quas et ad corporis voluptatibus. Iusto saepe fuga dolorum nisi ad repellat neque maxime.",
            "healthCheckUrl": "healthCheckUrl",
            "isOpenSource": true,
            "name": "Ergonomic Plastic Chicken 781 813",
            "status": "Submitted",
            "version": "version"
        }
    ],
    "talents": [
        {
            "name": "Unbranded Plastic Shoes 630 493",
            "shortDescription": "Ipsa eos minus debitis fugiat."
        }
    ],
    "contributors": [
        {
            "name": null,
            "address": null,
            "email": null,
            "phone": null
        }
    ],
    "longDescription": null,
    "name": "Rustic Granite Ball 210 663",
    "shortDescription": "Short description of the contribution"
}
```

ywkim312 commented:
> {
>     "capabilities": [
>         {
>             "apiBaseUrl": null,
>             "apiDocUrl": null,
>             "dataDeletionEndpointDetails": {
>                 "apiKey": "apiKey",
>                 "description": "description",
>                 "deletionEndpoint": "deletionEndpoint"
>             },
>             "deploymentDetails": {
>                 "authMethod": null,
>                 "databaseDetails": null,
>                 "environmentVariables": null,
>                 "location": "internal"
>             },
>             "description": "Cupiditate esse aliquid tenetur illum et consequuntur consequatur. Et non vel ratione dolore totam suscipit numquam nemo. In vitae sed consequatur sunt sit et omnis id quia.
>  
> Rerum dicta a sint tenetur voluptatum voluptate. Nesciunt labore neque. Ducimus sunt qui asperiores quaerat facilis iste in. Minima dicta molestiae est laboriosam aut non corporis consectetur. Sit aperiam et accusamus tempore atque nam occaecati. Id nesciunt molestiae beatae sed sequi laudantium autem.
>  
> Accusamus magnam veritatis eius. Enim a dolorem quasi dolor neque voluptas debitis. Culpa est ut vero facere possimus. Quia facere velit voluptates hic quas et ad corporis voluptatibus. Iusto saepe fuga dolorum nisi ad repellat neque maxime.",
>             "healthCheckUrl": "healthCheckUrl",
>             "isOpenSource": true,
>             "name": "Ergonomic Plastic Chicken 781 813",
>             "status": "Submitted",
>             "version": "version"
>         }
>     ],
>     "talents": [
>         {
>             "name": "Unbranded Plastic Shoes 630 493",
>             "shortDescription": "Ipsa eos minus debitis fugiat."
>         }
>     ],
>     "contributors": [
>         {
>             "name": null,
>             "address": null,
>             "email": null,
>             "phone": null
>         }
>     ],
>     "longDescription": null,
>     "name": "Rustic Granite Ball 210 663",
>     "shortDescription": "Short description of the contribution"
> }

I tested it and it was not a valid Json. The problem was a description part that had several carriage return. After I fixed it, it worked okay. Here is the one that I fixed.

{
	"capabilities": [{
		"apiBaseUrl": null,
		"apiDocUrl": null,
		"dataDeletionEndpointDetails": {
			"apiKey": "apiKey",
			"description": "description",
			"deletionEndpoint": "deletionEndpoint"
		},
		"deploymentDetails": {
			"authMethod": null,
			"databaseDetails": null,
			"environmentVariables": null,
			"location": "internal"
		},
		"description": "Cupiditate esse aliquid tenetur illum et consequuntur consequatur. Et non vel ratione dolore totam suscipit numquam nemo. In vitae sed consequatur sunt sit et omnis id quia. Rerum dicta a sint tenetur voluptatum voluptate.Nesciunt labore neque.Ducimus sunt qui asperiores quaerat facilis iste in .Minima dicta molestiae est laboriosam aut non corporis consectetur.Sit aperiam et accusamus tempore atque nam occaecati.Id nesciunt molestiae beatae sed sequi laudantium autem. Accusamus magnam veritatis eius.Enim a dolorem quasi dolor neque voluptas debitis.Culpa est ut vero facere possimus.Quia facere velit voluptates hic quas et ad corporis voluptatibus.Iusto saepe fuga dolorum nisi ad repellat neque maxime.",
		"healthCheckUrl": "healthCheckUrl",
		"isOpenSource": true,
		"name": "Ergonomic Plastic Chicken 781 813",
		"status": "Submitted",
		"version": "version"
	}],
	"talents": [{
		"name": "Unbranded Plastic Shoes 630 493",
		"shortDescription": "Ipsa eos minus debitis fugiat."
	}],
	"contributors": [{
		"name": null,
		"address": null,
		"email": null,
		"phone": null
	}],
	"longDescription": null,
	"name": "Rustic Granite Ball 210 663",
	"shortDescription": "Short description of the contribution"
}

sandeep-ps commented:
@ywkim312, Oops! Thanks for catching that. I have updated the Postman settings to use random single sentences.
